lib/pages/child/task_in_progress_page.dart

```diff
@@ -53,44 +53,50 @@ class _ChildTaskInProgressPageState extends State<ChildTaskInProgressPage> with
 					BlocProvider.of<TaskInstanceCubit>(context).loadTaskInstance();
 					isInitial = true;
 				}
-				return  Scaffold(
-					appBar: isInitial ? _getHeader(TaskInstanceLoaded(null, this.widget.initialPlanInstance)) : _getHeader(state),
-					body: isInitial ? Center(child: AppLoader())
-						: Padding(
-						padding: EdgeInsets.only(bottom: 0.0),
-						child: Stack(
-							children: _getAllCards(state)
-						)
-					),
-					extendBody: true,
-					bottomNavigationBar: isInitial ? SizedBox.shrink() : _getBottomNavBar(),
-					floatingActionButtonLocation: FloatingActionButtonLocation.endDocked,
-					floatingActionButton: isInitial ? SizedBox.shrink()
-						: Row(
-						mainAxisAlignment: MainAxisAlignment.spaceBetween,
-						children: <Widget>[
-							_getFab(true, state),
-							SlideTransition(
-								position: _taskListFabAnimation,
-								child: FloatingActionButton.extended(
-									label:
-									Row(
-										children: [
-											Icon(Icons.format_list_bulleted,),
-											Padding(
-												padding: const EdgeInsets.only(left: 8.0),
-												child: Text(AppLocales.of(context).translate('actions.return')),
-											),
-										],
-									),
-									onPressed: () => Navigator.of(context).pop(),
-									heroTag: "fabTasks",
-									backgroundColor: AppColors.childTaskColor,
-								),
-							),
-							_getFab(false, state)
-						],
-					),
+				return  WillPopScope(
+				  onWillPop: () async {
+						Navigator.of(context).pop({"plan" : state is TaskInstanceLoaded ? state.planInstance : null});
```

Comment:
Do we need this whole thing? `PlanInstanceCubit` is reloadable so it should update on it's own